manifests/plugin/default_hostgroup.pp

```diff
@@ -1,4 +1,20 @@
-# Installs foreman_default_hostgroup plugin
-class foreman::plugin::default_hostgroup {
-  foreman::plugin {'default_hostgroup': }
+# @summary This class installs the default_hostgroup plugin and optionally manages the configuration file
+#
+# @param hostgroups An array of hashes of hostgroup names and facts to add to the configuration
+#
+class foreman::plugin::default_hostgroup (
+  Array[Hash] $hostgroups = [],
+){
+  if empty($hostgroups) {
+    $config = undef
+    $config_file = undef
+  } else {
+    $config = template('foreman/default_hostgroup.yaml.erb')
+    $config_file = "${foreman::plugin_config_dir}/default_hostgroup.yaml"
+  }
+
+  foreman::plugin {'default_hostgroup':
+    config      => $config,
+    config_file => $config_file,
```

Comment:
You can set this unconditionally:
```suggestion
    config_file => "${foreman::plugin_config_dir}/default_hostgroup.yaml",
```

The file is only managed if the config is set:
https://github.com/theforeman/puppet-foreman/blob/9bb72cee31c3d7265f4f83c9dc222f7205848fed/manifests/plugin.pp#L46-L55

Reply:
Done